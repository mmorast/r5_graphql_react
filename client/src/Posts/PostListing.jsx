import React from 'react'
import PropTypes from 'prop-types'
import {Grid} from 'react-bootstrap'
import PostSummary from "./PostSummary";

const PostListing = props => {
  const {posts, ...rest} = props
  return (
    <Grid fluid={true}>
      {posts.map(post => <PostSummary key={`post-summary-${post.id}`} {...rest} post={post}/>)}
    </Grid>
  )
}

PostListing.propTypes = {
  viewer: PropTypes.object,
  posts: PropTypes.array.isRequired,
}

export default PostListing
