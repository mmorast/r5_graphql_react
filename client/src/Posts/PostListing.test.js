import React from 'react';
import {shallow} from 'enzyme';
import PostListing from './PostListing';

test("It should render the PostListing Component", () => {
  const posts = [
    {id: 1, title: 'one', body: 'post one'},
    {id: 2, title: 'two', body: 'post two'}
  ]
  const actual = shallow(<PostListing posts={posts}/>)
  expect(actual).toMatchSnapshot()
})
