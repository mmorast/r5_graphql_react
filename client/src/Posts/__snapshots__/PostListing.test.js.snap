// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render the PostListing Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
>
    <PostSummary
        post={
            Object {
                "body": "post one",
                "id": 1,
                "title": "one",
              }
        }
    />
    <PostSummary
        post={
            Object {
                "body": "post two",
                "id": 2,
                "title": "two",
              }
        }
    />
</Grid>,
  "nodes": Array [
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={true}
>
      <PostSummary
            post={
                  Object {
                        "body": "post one",
                        "id": 1,
                        "title": "one",
                      }
            }
      />
      <PostSummary
            post={
                  Object {
                        "body": "post two",
                        "id": 2,
                        "title": "two",
                      }
            }
      />
</Grid>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <PostListing
      posts={
            Array [
                  Object {
                    "body": "post one",
                    "id": 1,
                    "title": "one",
                  },
                  Object {
                    "body": "post two",
                    "id": 2,
                    "title": "two",
                  },
                ]
      }
/>,
    "_instance": null,
    "_newState": null,
    "_rendered": <Grid
      bsClass="container"
      componentClass="div"
      fluid={true}
>
      <PostSummary
            post={
                  Object {
                        "body": "post one",
                        "id": 1,
                        "title": "one",
                      }
            }
      />
      <PostSummary
            post={
                  Object {
                        "body": "post two",
                        "id": 2,
                        "title": "two",
                      }
            }
      />
</Grid>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PostListing
    posts={
        Array [
            Object {
              "body": "post one",
              "id": 1,
              "title": "one",
            },
            Object {
              "body": "post two",
              "id": 2,
              "title": "two",
            },
          ]
    }
/>,
}
`;
